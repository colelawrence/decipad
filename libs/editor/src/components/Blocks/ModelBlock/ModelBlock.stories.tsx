import React from 'react';
import { ModelBlock } from './ModelBlock.component';

export default {
  title: 'Editor/Blocks/Model Block',
  component: ModelBlock,
};

export const Default = (args: { text: string }) => (
  <ModelBlock
    leaf={{ attributes: {}, text: '' }}
    element={{ children: [{ text: '' }] }}
    text={{ text: '' }}
    attributes={{
      'data-slate-node': 'element',
      ref: null,
      'data-slate-leaf': true,
    }}
  >
    {args.text}
  </ModelBlock>
);

Default.args = {
  text: 'a = 10 apples',
};

Default.storyName = 'Model Block';
