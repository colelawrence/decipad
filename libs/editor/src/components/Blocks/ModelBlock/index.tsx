import styled from '@emotion/styled';
import { SlatePluginComponent } from '@udecode/slate-plugins';
import React from 'react';

const Element = styled('div')({
  borderRadius: '16px',
  padding: '24px',
  backgroundColor: 'rgba(240, 240, 242, 0.2)',
  border: '1px solid #F0F0F2',
  lineHeight: '2.5',
  margin: '8px 0',
  boxShadow: '0px 2px 24px -4px rgba(36, 36, 41, 0.06)',
  fontFamily: 'monospace',
  fontSize: '16px',
});

export const ModelBlock: SlatePluginComponent = ({ attributes, children }) => {
  return <Element {...attributes}>{children}</Element>;
};
