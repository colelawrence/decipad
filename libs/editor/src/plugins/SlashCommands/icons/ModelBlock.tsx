import { FC } from 'react';

export const ModelBlock = (): ReturnType<FC> => {
  return (
    <svg
      width="16"
      height="18"
      viewBox="0 0 16 18"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M15.1664 4.9772C15.1647 4.974 15.1634 4.97071 15.1615 4.96753C15.1608 4.96616 15.1598 4.96496 15.159 4.9636C15.0583 4.79075 14.9134 4.6479 14.739 4.54978L8.55155 1.06931C8.38296 0.97519 8.19308 0.925781 8 0.925781C7.80692 0.925781 7.61704 0.97519 7.44845 1.06931L1.26095 4.54978C1.08602 4.64827 0.940645 4.79182 0.839937 4.96549C0.838838 4.96739 0.837569 4.96909 0.836504 4.97102C0.834788 4.97409 0.833551 4.97726 0.831904 4.98034C0.73742 5.14825 0.687694 5.33764 0.6875 5.53031V12.4661C0.687648 12.6657 0.740822 12.8617 0.841582 13.034C0.942342 13.2062 1.08707 13.3487 1.26095 13.4466L7.44849 16.9271C7.60454 17.0144 7.77918 17.0631 7.95787 17.0693C7.97044 17.0702 7.98297 17.0714 7.99578 17.0715C7.99746 17.0716 7.99914 17.0716 8.00082 17.0716C8.0185 17.0716 8.03591 17.0706 8.05318 17.069C8.22809 17.0612 8.39875 17.0126 8.55155 16.9271L14.739 13.4466C14.9129 13.3487 15.0577 13.2062 15.1584 13.034C15.2592 12.8617 15.3124 12.6657 15.3125 12.4661V5.53031C15.3123 5.33642 15.262 5.14588 15.1664 4.9772ZM8.064 8.35281L2.40164 5.19891L8 2.04984L13.607 5.20377L8.064 8.35281ZM14.1875 12.4661L8.57101 15.6254L8.62636 9.3272L14.1875 6.16782V12.4661Z"
        fill="#121214"
      />
    </svg>
  );
};
