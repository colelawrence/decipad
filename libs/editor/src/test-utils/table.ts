import {
  ELEMENT_TABLE,
  ELEMENT_TH,
  ELEMENT_TR,
  ELEMENT_TD,
} from '@udecode/plate';
import {
  InteractiveTable,
  TableCellType,
  Td,
  Th,
} from '../plugins/InteractiveTable/table';
import {
  ELEMENT_TABLE_CAPTION,
  ELEMENT_THEAD,
  ELEMENT_HEAD_TR,
  ELEMENT_TBODY,
} from '../utils/elementTypes';

/**
 * Creates a row with 1 named column and {@param numberOfRows} rows of non-empty cells.
 */
export const createTable = (numberOfRows: number): InteractiveTable => ({
  type: ELEMENT_TABLE,
  children: [
    {
      type: ELEMENT_TABLE_CAPTION,
      children: [{ text: 'Table' }],
    },
    {
      type: ELEMENT_THEAD,
      children: [
        {
          type: ELEMENT_HEAD_TR,
          children: [
            {
              type: ELEMENT_TH,
              attributes: { cellType: 'string' },
              children: [{ text: 'Col 1' }],
            },
          ],
        },
      ],
    },
    {
      type: ELEMENT_TBODY,
      children: Array.from({ length: numberOfRows }, (_, i) => ({
        type: ELEMENT_TR,
        children: [
          {
            type: ELEMENT_TD,
            attributes: { cellType: 'string' },
            children: [{ text: `Col 1 Row ${i + 1}` }],
          },
        ],
      })),
    },
  ],
});

export const createTableWithTypeAndData = (
  cellType: TableCellType,
  data: string[]
): InteractiveTable => ({
  type: ELEMENT_TABLE,
  children: [
    {
      type: ELEMENT_TABLE_CAPTION,
      children: [{ text: 'Table' }],
    },
    {
      type: ELEMENT_THEAD,
      children: [
        {
          type: ELEMENT_HEAD_TR,
          children: [
            {
              type: ELEMENT_TH,
              attributes: { cellType },
              children: [{ text: 'Col 1' }],
            },
          ],
        },
      ],
    },
    {
      type: ELEMENT_TBODY,
      children: data.map((text) => ({
        type: ELEMENT_TR,
        children: [
          {
            type: ELEMENT_TD,
            attributes: { cellType },
            children: [{ text }],
          },
        ],
      })),
    },
  ],
});

export const printTable = ({
  children: [_, thead, tbody],
}: InteractiveTable): string[] => {
  const tdsToStrings = (tds: (Th | Td)[]) =>
    tds.map((td) => td.children[0].text).join(' | ');

  return [
    tdsToStrings(thead.children[0].children),
    ...tbody.children.map((tr) => tdsToStrings(tr.children)),
  ];
};
