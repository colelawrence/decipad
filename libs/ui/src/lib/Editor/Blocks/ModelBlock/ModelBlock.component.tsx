import { css } from '@emotion/react';
import { PlatePluginComponent } from '@udecode/plate';
import { Result } from './Result.component';

const modelBlockStyles = css({
  borderRadius: '16px',
  padding: '24px',
  backgroundColor: 'rgba(240, 240, 242, 0.2)',
  border: '1px solid #F0F0F2',
  lineHeight: '2.5',
  margin: '16px 0',
  fontFamily: 'monospace',
  boxShadow: '0px 2px 24px -4px rgba(36, 36, 41, 0.06)',
});

export const ModelBlockElement: PlatePluginComponent = ({
  attributes,
  children,
  element,
}) => {
  const blockId = (element as any).id ?? '';

  return (
    <div spellCheck={false}>
      <pre css={modelBlockStyles} {...attributes}>
        {children}
      </pre>
      {blockId && <Result blockId={blockId} />}
    </div>
  );
};
