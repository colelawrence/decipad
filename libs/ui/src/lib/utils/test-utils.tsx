import { ChakraProvider } from '@chakra-ui/react';
import { render, RenderOptions } from '@testing-library/react';
import React from 'react';

import { Type } from '@decipad/language';
import { theme, ResultsContextProvider } from '../..';

const results = {
  test: {
    type: Type.build({
      type: 'number',
      unit: {
        unit: 'apples',
      },
    } as any),
    value: [20],
    errors: [],
  },
  noResult: null,
};

const AllTheProviders: React.FC = ({ children }) => {
  return (
    <ChakraProvider resetCSS theme={theme}>
      <ResultsContextProvider value={results}>
        {children}
      </ResultsContextProvider>
    </ChakraProvider>
  );
};

const customRender = (
  ui: JSX.Element,
  options?: Omit<RenderOptions, 'queries'>
) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';
// override render method
export { customRender as render };
