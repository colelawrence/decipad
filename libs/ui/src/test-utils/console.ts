export const mockConsoleWarn = () => {
  const originalConsoleWarn = console.warn;
  const mockConsoleWarn: jest.MockedFunction<typeof console.warn> = jest.fn();

  beforeEach(() => {
    console.warn = jest.fn();
    mockConsoleWarn.mockClear();
  });
  afterEach(() => {
    console.warn = originalConsoleWarn;
  });

  return mockConsoleWarn;
};
