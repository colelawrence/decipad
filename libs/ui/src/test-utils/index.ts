export * from './menu';
export * from './language';
