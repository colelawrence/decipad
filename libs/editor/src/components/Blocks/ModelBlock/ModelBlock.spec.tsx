import React from 'react';
import { render } from 'test-utils';
import { Default } from './ModelBlock.stories';

describe('ModelBlock Block', () => {
  it('matches snapshot', () => {
    const { container } = render(<Default text="a = 10 apples" />);
    expect(container).toMatchSnapshot();
  });
});
