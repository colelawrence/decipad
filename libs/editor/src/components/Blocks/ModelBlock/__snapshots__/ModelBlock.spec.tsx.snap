// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModelBlock Block matches snapshot 1`] = `
.emotion-0 {
  border-radius: 16px;
  padding: 24px;
  background-color: rgba(240, 240, 242, 0.2);
  border: 1px solid #F0F0F2;
  line-height: 2.5;
  margin: 8px 0;
  box-shadow: 0px 2px 24px -4px rgba(36, 36, 41, 0.06);
  font-family: monospace;
  font-size: 16px;
}

<div>
  <div
    class="emotion-0"
    data-slate-leaf="true"
    data-slate-node="element"
  >
    a = 10 apples
  </div>
</div>
`;
